packages/block/src/block.ts

```diff
@@ -208,7 +221,18 @@ export class Block {
       }))
       ?.map(Withdrawal.fromWithdrawalData)
 
-    return new Block(header, transactions, uncleHeaders, withdrawals, opts)
+    // executionWitness are not part of the EL fetched blocks via eth_ bodies method
+    // they are currently only available via the engine api constructed blocks
+    let executionWitness
+    if (executionWitnessBytes !== undefined) {
+      executionWitness = JSON.parse(bytesToUtf8(RLP.decode(executionWitnessBytes) as Uint8Array))
+    } else {
+      // don't assign default witness if eip 6800 is implemented as it leads to incorrect
+      // assumptions while executing the block. if not present in input implies its unavailable
+      executionWitness = null
```

Comment:
I am not completely getting the `null` vs `undefined` case here. Is `null` only for the 6800 activated case? If there is no distinction: can we use only `undefined` and skip the `null` case (as we do for withdrawals)?

If there is a distinction: can we for the non-6800 case only use `undefined` (which is not the case here)?

Reply:
yes the distinction is there:

null means there is a non default witness and a default witness shouldn't be assigned in the constructor
undefined means that constructor can assign a default witness which is handy in block constructions for tests etc 

at this point we don't know the the common's hardfork because there could be setHardfork params to it. but what it esentially says to the block constructor is if the construction comes from  fromValues don't assign a deafult witness

Reply:
Ok, thanks for the clarification

Reply:
Update: ok, I am also not completely overseeing the whole "working together", but we should minimally make sure that for the (all) 6800-deactivated cases there is never a default witness assigned to the block (not sure if this is still the case).

Even if "only internal" this would be wrong internal data and could (too easy) lead to side effects e.g. in serialization/output (JSON RPC) cases or whatever (accidentally also printing out witnesses).

Reply:
yes we don't assign default witness if 6800 is not activated (see L414)